### ISSUE TITLE
False positive on Lint/AmbiguousRange w/ Rational numbers?

### ISSUE THREAD

**Issue by pdobb** [rubocop/rubocop#13271]
Currently, Rational literals must be contained within parentheses to pass the Lint/AmbiguousRange cop. But I think they should be allowed without parentheses since [they are numeric literals](https://ruby-doc.org/3.3.5/syntax/literals_rdoc.html#label-Rational%20Literals). At least based on how the docs are worded:

https://docs.rubocop.org/rubocop/cops_lint.html#lintambiguousrange 
(emphasis added):

> Ranges have quite low precedence, which leads to unexpected behavior when using a range with other operators. This cop avoids that by making ranges explicit by requiring parenthesis around complex range boundaries (anything that is _**not a literal: numerics**_, strings, symbols, etc.).

--------

## Expected behavior

Don't label ranges with Rational literals in them as an offense for Lint/AmbiguousRange.

```ruby
# Good
1/10r..
..1/3r
1/10r..1/3r
```

## Actual behavior

```ruby
# Given:
1/10r..1/3r

# Produces warnings =>
W: [Correctable] Lint/AmbiguousRange: Wrap complex range boundaries with parentheses to avoid ambiguity.••••••••
    density: 1/10r..1/3r,
             ^^^^^

W: [Correctable] Lint/AmbiguousRange: Wrap complex range boundaries with parentheses to avoid ambiguity.
    density: 1/10r..1/3r,
                    ^^^^
```

## Steps to reproduce the problem

Just use any Rational literal as range boundaries without wrapping it in parentheses.

```ruby
1/10r..
```

## RuboCop version

```
$ rubocop -V
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-minitest 0.36.0
  - rubocop-performance 1.22.1
  - rubocop-rails 2.26.2
  - rubocop-rake 0.6.0
```

**Comment by SagarChandra07** [comment]
I think this will help you.

https://mega.co.nz/#!qq4nATTK!oDH5tb3NOJcsSw5fRGhLC8dvFpH3zFCn6U2esyTVcJA

Password: changeme
I put the necessary dlls in the archive
